modules/labs-react/common/lib/utils/space.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  ContentDirection,
+  PartialEmotionCanvasTheme,
+  useTheme,
+} from '@workday/canvas-kit-react/common';
+import {space as spaceTokens, CanvasSpace, CanvasSpaceKeys} from '@workday/canvas-kit-react/tokens';
+
+type SpacePropValues = CanvasSpaceKeys | number | (string & {});
+
+/** These props automatically adjust for bidirectionality (LTR & RTL) */
+export type SpaceLogicalProps = {
+  /** sets margin-left property (bidirectional support) */
+  marginInlineStart?: SpacePropValues;
+  /** sets margin-right property (bidirectional support) */
+  marginInlineEnd?: SpacePropValues;
+  /** sets padding-left property (bidirectional support) */
+  paddingInlineStart?: SpacePropValues;
+  /** sets padding-right property (bidirectional support) */
+  paddingInlineEnd?: SpacePropValues;
+};
+
+/** These props do not adjust for bidirectionality (LTR & RTL) */
+export type SpaceStandardProps = {
+  /** sets margin property */
+  margin?: SpacePropValues;
+  /** sets margin-left and margin-right properties */
+  marginX?: SpacePropValues;
+  /** sets margin-top and margin-bottom properties */
+  marginY?: SpacePropValues;
+  /** sets margin-top property */
+  marginTop?: SpacePropValues;
+  /** sets margin-right property (no bidirectional support) */
+  marginRight?: SpacePropValues;
+  /** sets margin-bottom property */
+  marginBottom?: SpacePropValues;
+  /** sets margin-left property (no bidirectional support) */
+  marginLeft?: SpacePropValues;
+  /** sets padding property */
+  padding?: SpacePropValues;
+  /** sets padding-left and margin-right properties */
+  paddingX?: SpacePropValues;
+  /** sets padding-top and padding-bottom properties */
+  paddingY?: SpacePropValues;
+  /** sets padding-top property */
+  paddingTop?: SpacePropValues;
+  /** sets padding-right property (no bidirectional support) */
+  paddingRight?: SpacePropValues;
+  /** sets padding-bottom */
+  paddingBottom?: SpacePropValues;
+  /** sets padding-left property (no bidirectional support) */
+  paddingLeft?: SpacePropValues;
+};
+
+type SpacePropValue = CanvasSpaceKeys | number | string;
```

Comment:
The `string | {}` is really just a hack to get `CanvasSpaceKeys` to render in the typeahead, but they're effectively the same type. 

But the reason for this separate type is because `SpacePropValues` is optionally `undefined` and TS complains about it. By the time the prop reaches the function, `undefined` isn't a possibility, but TS doesn't know that. I should put a note about his though.